R/derive_params_growth_age.R

```diff
@@ -0,0 +1,285 @@
+#'
+#'   Expected Values: `M`, `F`
+#'
+#' @param age Current Age
+#'
+#'   A numeric vector is expected. Note that this is the actual age at the current visit.
+#'
+#' @param age_unit Age Unit
+#
+#'   A character vector is expected.
+#'
+#'   Expected values: 'days' 'weeks' 'months'
+#'
+#' @param meta_criteria Metadata dataset
+#'
+#'   A metadata dataset with the following expected variables:
+#'   `AGE`, `AGEU`, `SEX`, `L`, `M`, `S`
+#'
+#'   The dataset can be derived from CDC/WHO or user-defined datasets.
+#'   The CDC/WHO growth chart metadata datasets are available in the package and will
+#'   require small modifications.
+#'   * `AGE` - Age
+#'   * `AGEU` - Age Unit
+#'   * `SEX` - Sex
+#'   * `L` - Power in the Box-Cox transformation to normality
+#'   * `M` - Median
+#'   * `S` - Coefficient of variation
+#'
+#' @param parameter Anthropometric measurement parameter to calculate z-score or percentile
+#'
+#'   A condition is expected with the input dataset `VSTESTCD`/`PARAMCD`
+#'   for which we want growth derivations:
+#'
+#'   e.g. `parameter = VSTESTCD == "WEIGHT"`.
+#'
+#'   There is CDC/WHO metadata available for Height, Weight, BMI, and Head Circumference available
+#'   in the `admiralpeds` package.
+#'
+#' @param analysis_var Variable containing anthropometric measurement
+#'
+#' A numeric vector is expected, e.g. `AVAL`, `VSSTRESN`
+#'
+#' @param bmi_cdc_correction Extended CDC BMI-for-age correction
+#'
+#'  A logical scalar, e.g. `TRUE`/`FALSE` is expected.
+#'  CDC developed extended percentiles (>95%) to monitor high BMI values,
+#'  if set to `TRUE` the CDC's correction is applied.
+#'
+#' @param set_values_to_sds Variables to be set for Z-Scores
+#'
+#'  The specified variables are set to the specified values for the new
+#'  observations. For example,
+#'   `set_values_to_sds(exprs(PARAMCD = “BMIASDS”, PARAM = “BMI-for-age z-score”))`
+#'  defines the parameter code and parameter.
+#'
+#' *Permitted Values*: List of variable-value pairs
+#'
+#'  If left as default value, `NULL`, then parameter not derived in output dataset
+#'
+#' @param set_values_to_pctl Variables to be set for Percentile
+#'
+#'  The specified variables are set to the specified values for the new
+#'  observations. For example,
+#'   `set_values_to_pctl(exprs(PARAMCD = “BMIAPCTL”, PARAM = “BMI-for-age percentile”))`
+#'  defines the parameter code and parameter.
+#'
+#' *Permitted Values*: List of variable-value pair
+#'
+#'  If left as default value, `NULL`, then parameter not derived in output dataset
+#'
+#' @return The input dataset additional records with the new parameter added.
+#'
+#'
+#' @family der_prm_bds_vs
+#'
+#' @keywords der_prm_bds_vs
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr)
+#' library(lubridate)
+#' library(rlang)
+#' library(admiral)
+#'
+#' advs <- dm_peds %>%
+#'   select(USUBJID, BRTHDTC, SEX) %>%
+#'   right_join(., vs_peds, by = "USUBJID") %>%
+#'   filter(USUBJID != "PEDS-1010") %>%
```

Comment:
same comments as for the by height function - the test data has changed here now to make patients consistent with pharmaversesdtm so no "PEDS-1010" for example.